pkg/chunk/purger/tombstones.go

```diff
@@ -16,6 +20,35 @@ import (
 
 const tombstonesReloadDuration = 5 * time.Minute
 
+type tombstonesLoaderMetrics struct {
+	cacheGenLoadFailures       prometheus.Gauge
```

Comment:
I did it this way to help with alerting. If we keep incrementing it then we won't be able to figure out the state of the metric and decide whether the issue has resolved or not. I might be wrong though here. Do you think it would be easy to determine the alert state if I do not reset it and keep incrementing whenever we fail to load the cache gen?

Reply:
In our alerts we can use a rate function which will let us see the state at any given time. There should be no need to reset it. Just increment it every time a failure occurs.

If we keep it as a gauge, that leaves open the potential to miss failures if the gauge resets before we scrape the instance.

Reply:
That makes sense. Thanks!